### PR TITLE
split: implement remaining -n variants and pass GNU tests/split/fail.sh

### LINKED ISSUE #5033: `split -1 in` should be accepted
```
touch in
./target/debug/coreutils split -1 in
```
=> 
```
error: unexpected argument '-1' found

  tip: to pass '-1' as a value, use '-- -1'
```

It should be equivalent to `-a 1`

### PR DESCRIPTION
**Pull request by zhitkoff**
A draft PR to implement 2 missing variants of --number (-n) option for `split`
```
K/N
r/K/N
```
As well as passing GNU `tests/split/fail.sh`

NOTE: Includes code changes from PR #5222, so this one can be merged instead of #5222 

Fixes #5033

### CODE REVIEW

**Comment by zhitkoff** [comment]
@tertsdiepraam @sylvestre this is rather big one, but all of that had to be included to pass GNU test `tests/split/fail.sh`
It can replace the PR #5252 for `split -1 in` as it includes all its functionality (also required to pass that GNU test)
Please let me know how you would like to proceed between these two PRs

File: src/uu/split/src/split.rs [hunk 1]
```diff
@@ -52,14 +54,127 @@ const AFTER_HELP: &str = help_section!("after help", "split.md");
...
     }
 }
 
+/// Extract obsolete shorthand (if any) for specifying lines in following scenarios (and similar)
+/// `split -22 file` would mean `split -l 22 file`
+/// `split -2de file` would mean `split -l 2 -d -e file`
+/// `split -x300e file` would mean `split -x -l 300 -e file`
+/// `split -x300e -22 file` would mean `split -x -e -l 22 file` (last obsolete lines option wins)
+/// following GNU `split` behavior
+fn handle_obsolete(args: impl uucore::Args) -> (Vec<OsString>, Option<String>) {
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:74]: This function is a bit too long
What about something like:
```
fn handle_obsolete(args: impl uucore::Args) -> (Vec<OsString>, Option<String>) {
    let mut obs_lines = None;
    let mut preceding_long_opt_req_value = false;
    let mut preceding_short_opt_req_value = false;

    let filtered_args = args
        .filter_map(|os_slice| filter_args(
            os_slice,
            &mut obs_lines,
            &mut preceding_long_opt_req_value,
            &mut preceding_short_opt_req_value,
        ))
        .collect();

    (filtered_args, obs_lines)
}

fn filter_args(
    os_slice: OsString,
    obs_lines: &mut Option<String>,
    preceding_long_opt_req_value: &mut bool,
    preceding_short_opt_req_value: &mut bool,
) -> Option<OsString> {
    let filter: Option<OsString>;
    if let Some(slice) = os_slice.to_str() {
        if should_extract_obs_lines(slice, preceding_long_opt_req_value, preceding_short_opt_req_value) {
            filter = handle_extract_obs_lines(slice, obs_lines);
        } else {
            filter = Some(OsString::from(slice));
        }
        handle_preceding_options(slice, preceding_long_opt_req_value, preceding_short_opt_req_value);
    } else {
        filter = Some(os_slice);
    }
    filter
}

fn should_extract_obs_lines(slice: &str, preceding_long_opt_req_value: &bool, preceding_short_opt_req_value: &bool) -> bool {
    slice.starts_with('-')
        && !slice.starts_with("--")
        && !preceding_long_opt_req_value
        && !preceding_short_opt_req_value
        && !slice.starts_with("-a")
        && !slice.starts_with("-b")
        && !slice.starts_with("-C")
        && !slice.starts_with("-l")
        && !slice.starts_with("-n")
}

fn handle_extract_obs_lines(slice: &str, obs_lines: &mut Option<String>) -> Option<OsString> {
    // ... (existing logic for extracting obsolete lines)
}

fn handle_preceding_options(slice: &str, preceding_long_opt_req_value: &mut bool, preceding_short_opt_req_value: &mut bool) {
    // ... (existing logic for handling preceding options)
}
```

File: src/uu/split/src/split.rs [hunk 2]
```diff
@@ -292,50 +423,62 @@ impl NumberType {
...
                 } else {
                     Err(NumberTypeError::NumberOfChunks(s.to_string()))
                 }
             }
+            [k_str, n_str] if !k_str.starts_with('l') && !k_str.starts_with('r') => {
+                let num_chunks = parse_size(n_str)
+                    .map_err(|_| NumberTypeError::NumberOfChunks(n_str.to_string()))?;
+                let chunk_number = parse_size(k_str)
+                    .map_err(|_| NumberTypeError::ChunkNumber(k_str.to_string()))?;
+                if chunk_number > num_chunks || chunk_number == 0 {
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:449]: you have the same check 3 times
please using an helper function like:

fn is_invalid_chunk(chunk_number: i32, num_chunks: i32) -> bool {
    chunk_number > num_chunks || chunk_number == 0
}

**Comment by sylvestre** [comment]
besides these minor two comments, great work :)

File: src/uu/split/src/split.rs (hunk 1 above)
Comment (zhitkoff) [reply src/uu/split/src/split.rs:74]: yep, will refactor

File: src/uu/split/src/split.rs (hunk 2 above)
Comment (zhitkoff) [reply src/uu/split/src/split.rs]: thanks, will do

**Comment by zhitkoff** [comment]
> besides these minor two comments, great work :)

That you! :) I have refactored it a bit following your suggestions

**Comment by zhitkoff** [comment]
@sylvestre could you please let me know if you have any other suggestions/comments before this PR can be merged?

**Comment by sylvestre** [comment]
it looks great. @tertsdiepraam don't hesitate if you want to do a post landing review

**Pull request merged**